actionpack/lib/action_dispatch/routing/mapper.rb

```diff
@@ -355,7 +355,7 @@ def split_to(to)
 
           def add_controller_module(controller, modyoule)
             if modyoule && !controller.is_a?(Regexp)
-              if controller && controller.start_with?("/")
+              if %r{\A/}.match?(controller)
```

Comment:
This will reintroduce the deprecation warning fixed in https://github.com/rails/rails/pull/37504.

We need to avoid the comparison when `controller` is nil:

```suggestion
              if controller && %r{\A/}.match?(controller)
```